shell/browser/api/electron_api_base_window.cc

```diff
@@ -177,8 +177,15 @@ void BaseWindow::OnWindowClosed() {
       FROM_HERE, GetDestroyClosure());
 }
 
-void BaseWindow::OnWindowEndSession() {
-  Emit("session-end");
+void BaseWindow::OnWindowQueryEndSession(const std::vector<std::string> reasons,
+                                         bool* prevent_default) {
+  if (Emit("query-session-end", reasons)) {
+    *prevent_default = true;
+  }
+}
+
+void BaseWindow::OnWindowEndSession(const std::vector<std::string> reasons) {
+  Emit("session-end", reasons);
```

Comment:
Pass by reference. Please also update other interfaces where `reasons` is passed in.
```suggestion
void BaseWindow::OnWindowQueryEndSession(const std::vector<std::string>& reasons,
                                         bool* prevent_default) {
  if (Emit("query-session-end", reasons)) {
    *prevent_default = true;
  }
}

void BaseWindow::OnWindowEndSession(const std::vector<std::string>& reasons) {
  Emit("session-end", reasons);
```